pkgs/build-support/fetchurl/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
+{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
```

Comment:
I like the idea of a fallback to pgp but that would be incredibly hard to
pull off given the variability of how signatures are stored if they are
generated at all.

Regardless, I think this is worth adding given the minimal overhead and it
is technically more correct to validate the certs. It is what an end user
would expect.

On Sun, May 31, 2015, 01:07 Vladimír Čunát <notifications@github.com> wrote:

> In pkgs/build-support/fetchurl/default.nix
> <https://github.com/NixOS/nixpkgs/pull/8082#discussion_r31387968>:
>
> > @@ -1,4 +1,4 @@
> > -{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
> > +{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
>
> I see. I suspect the main part of performance loss is due to extra latency
> of https handshake, but according to docs it seems that would happen
> *even* with --insecure with http*s*, so it's more about whether to prefer
> https URLs over http ones.
>
> I already thought before: it would be nice if fetching was done over plain
> http/ftp/... by default, but failed-hash ones would be re-checked securely
> afterward (ideally by PGP, or at least https re-fetch).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/8082/files#r31387968>.
>